R/slide.R

```diff
@@ -288,8 +288,26 @@ epi_slide <- function(
       dplyr::count(.data$time_value) %>%
       `[[`("n")
 
-    slide_values <- vctrs::list_unchop(slide_values_list)
+    if (length(slide_values_list) == 0L) {
+      # We don't know what .ptype we should be outputting, and we won't try to
+      # infer it by running a dummy computation. We should just output something
+      # that will combine well with what computations exist. In some edge cases
+      # (zero rows in .x, zero .ref_time_values) we may end up just not adding
+      # any columns, but those edge cases are currently explicitly handled
+      # earlier (outputting zero columns and aborting, respectively).
+
+      # To combine well, we want something of a "super"-.ptype of all possible
+      # values. `NULL` almost works but can't be `vec_rep`'d. We'll use a 0-col
+      # data.frame instead, but will have to ensure it's unpacked into its 0
+      # columns in case other computations return vectors by introducing a
+      # .group_new_col_name.
+      .group_new_col_name <- NULL
+      slide_values_list <- vctrs::new_list_of(slide_values_list, data.frame())
+    } else {
+      .group_new_col_name <- .new_col_name
+    }
```

Comment:
- Zero row .x here is actually already handled in epi_slide entry, I didn't touch it --- we just don't add columns.  Does #519 change this?
- `ref_time_values` not in `geo`'s time_values bounds, `.all_rows = FALSE` --- I think the behaviors should already be compatible
- `ref_time_values` not in `geo`'s time_values bounds, `.all_rows = TRUE` --- there might be a difference here.  The approach here will appropriately fill in with NAs (assuming we get at least one "real" result row from any group, which I believe should be guaranteed if .x has at least 1 row?); what does #519 do?

Reply:
> Zero row .x here is actually already handled in epi_slide entry, I didn't touch it --- we just don't add columns. Does 519 change this?

I wrote `.x` because your comment above refers to `.x`, even though you probably mean `.data_group`. I also mean `.data_group`.

> ref_time_values not in geo's time_values bounds, .all_rows = TRUE --- there might be a difference here. The approach here will appropriately fill in with NAs (assuming we get at least one "real" result row from any group, which I believe should be guaranteed if .x has at least 1 row?); what does 519 do?

```r
  # If the data group does not contain any of the reference time values, return
  # the original .data_group without slide columns and let bind_rows at the end
  # of group_modify handle filling the empty data frame with NA values.
  if (length(available_ref_time_values) == 0L) {
    if (.all_rows) {
      return(.data_group)
    }
    return(.data_group %>% filter(FALSE))
  }
```

It returns the `.data_group` and the bind_rows on the outside handles filling the NAs.

Reply:
I meant `.x`, not `.data_group`.  We have different behavior if `.x` is 0 rows vs. if just some `.data_group`s have 0 rows.  From your extracts above, though, I think 519 & this have compatible results.

Side note: the explicit 0-row `.x` handling on `epi_slide` entry seems redundant if we use 519's approach... and maybe also the one in this PR.  So we might be able to delete that.

Reply:
Alright, I'll just need to pay attention to this when merging to make sure there's no weirdness.